src/app.js

```diff
@@ -113,67 +113,57 @@ export const challenge = (wsecret) => (req, res, next) => {
 };
 
 // Create Express Web app
-export const webapp = (appId, secret, wsecret, cb) => {
-  // Authenticate the app and get an OAuth token
-  oauth.run(appId, secret, (err, token) => {
-    if(err) {
-      cb(err);
-      return;
-    }
+export const webapp = async (appId, secret, wsecret) => {
 
-    // Return the Express Web app
-    cb(null, express()
+  // Authenticate the app and get an OAuth token
+  const token = await oauth.run(appId, secret);
 
-      // Configure Express route for the app Webhook
-      .post('/echo',
+  // Configure Express route for the app Webhook
+  return express().post('/echo',
 
-        // Verify Watson Work request signature and parse request body
-        bparser.json({
-          type: '*/*',
-          verify: verify(wsecret)
-        }),
+    // Verify Watson Work request signature and parse request body
+    bparser.json({
+      type: '*/*',
+      verify: verify(wsecret)
+    }),
 
-        // Handle Watson Work Webhook challenge requests
-        challenge(wsecret),
+    // Handle Watson Work Webhook challenge requests
+    challenge(wsecret),
 
-        // Handle Watson Work messages
-        echo(appId, token)));
-  });
+    // Handle Watson Work messages
+    echo(appId, token));
 };
 
 // App main entry point
-const main = (argv, env, cb) => {
-  // Create Express Web app
-  webapp(
-    env.ECHO_APP_ID, env.ECHO_APP_SECRET,
-    env.ECHO_WEBHOOK_SECRET, (err, app) => {
-      if(err) {
-        cb(err);
-        return;
-      }
-
-      if(env.PORT) {
-        // In a hosting environment like Bluemix for example, HTTPS is
-        // handled by a reverse proxy in front of the app, just listen
-        // on the configured HTTP port
-        log('HTTP server listening on port %d', env.PORT);
-        http.createServer(app).listen(env.PORT, cb);
-      }
-
-      else
-        // Listen on the configured HTTPS port, default to 443
-        ssl.conf(env, (err, conf) => {
-          if(err) {
-            cb(err);
-            return;
-          }
-          const port = env.SSLPORT || 443;
-          log('HTTPS server listening on port %d', port);
-          https.createServer(conf, app).listen(port, cb);
-        });
-    });
+const main = async (argv, env, cb) => {
+  try {
+    // Create Express Web app
+    const app = await webapp(
+      env.ECHO_APP_ID, env.ECHO_APP_SECRET,
+      env.ECHO_WEBHOOK_SECRET); 
+    cb(null);
```

Comment:
I don't think there's a need for a callback here as this function was converted to async.